assets/agenda/components/AgendaPreviewPlanning.tsx

```diff
@@ -14,93 +15,203 @@ interface IOwnProps {
+                        ) : (
+                            this.props.planningItems?.map((planningItem: any) => {
+                                const isExpanded = expandedPlanningItems[planningItem._id] || false;
+                                return (
+                                    <div
+                                        key={planningItem.guid}
+                                        className={`agenda-planning__preview ${
+                                            isExpanded
+                                                ? 'agenda-planning__preview--expanded'
+                                                : ''
+                                        }`}
+                                    >
+                                        <div
+                                            className="agenda-planning__preview-header"
+                                            onClick={() =>
+                                                this.toggleExpanded(planningItem._id)
+                                            }
+                                        >
+                                            <i
+                                                className={`icon-small--arrow-down ${
+                                                    isExpanded ? '' : 'rotate-90-ccw'
+                                                }`}
+                                            />
+                                            <span>{planningItem.headline || 'No Headline'}</span>
+                                        </div>
+                                        {isExpanded && (
+                                            <AgendaPreviewCoverages
+                                                key={planningItem.guid}
+                                                item={item}
+                                                plan={planningItem}
+                                                wireItems={wireItems}
+                                                actions={coverageActions}
+                                                user={user}
+                                                previewGroup={previewGroup}
+                                                previewConfig={previewConfig}
+                                            />
+                                        )}
+                                    </div>
+                                );
+                            })
+                        )}
                     </div>
                 </div>
-            )}
-        </React.Fragment>
-    );
+            </React.Fragment>
+        );
+    }
 }
 
-const mapStateToProps = (state: IAgendaState): IReduxStateProps => ({
-    wireItems: state.agenda.agendaWireItems || [],
-});
+const mapStateToProps = (state: any, ownProps: any) => {
+    const planningIds = ownProps.item.planning_ids || [];
+    return {
+        planningItems: planningIds.map((eventId: string) => state.itemsById[eventId]),
```

Comment:
Again here, try not to construct new values in `mapStateToProps`, as React will see this as a new value every time. Either construct these in the render function, or create a redux selector that memoizes the results based on provided args